src/bscPlugin/validation/ScopeValidator.spec.ts

```diff
@@ -2239,6 +2239,95 @@ describe('ScopeValidator', () => {
         });
     });
 
+    describe('expectedReturnStatement', () => {
```

Comment:
We need tests for `sub as <returnType>` as well. This is valid code and works:
```brighterscript
sub getName() as string
  return "bob"
end sub
```

Missing return in `sub as <returnType>`  crashes like function.

![image](https://github.com/user-attachments/assets/72885006-25d2-4c7f-9282-5e81f56d7c3c)

![image](https://github.com/user-attachments/assets/854024f3-6f70-4b78-ad6c-22732239cb62)